server/app/lib/utils/cloud_forms/provider.rb

```diff
@@ -64,6 +64,25 @@ def self.add(cfme_ip, provider_params)
         }
         request_data = new_provider_form.request_data
         provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)
+
+        show_list = agent.get("https://#{cfme_ip}/host/show_list", [], agent.page.uri)
+        while show_list.links_with(:href => %r'/host/show/1r', :text => "\n\t\t\t\t\n\t\t\t\t").count < provider_params[:hypervisors].count
```

Comment:
I assume this while loop is waiting for the RHEV hypervisors to show up in CFME, so we need to guard against us running before CFME has finished the setup of RHEV.

You could do something where you allow this while loop to run for say ~20 minutes, or 40 iterations with your 30 sec sleep.

1) Declare a 'counter' variable outside of the while loop, initialize it to zero.
2) In the while loop, check if counter > 40, if it is greater than 20 raise an exception, else continue to sleep(30) and bump counter+=1